Admin/AdminHelper.php

```diff
@@ -201,4 +229,105 @@ public function camelize($property)
     {
         return BaseFieldDescription::camelize($property);
     }
+
+    /**
+     * @param AdminInterface $admin
+     * @param array          $elements
+     *
+     * @return string
+     */
+    protected function entityClassNameFinder(AdminInterface $admin, $elements)
+    {
+        $element = array_shift($elements);
+        $associationAdmin = $admin->getFormFieldDescription($element)->getAssociationAdmin();
+        if (count($elements) == 0) {
+            return $associationAdmin->getClass();
+        } else {
+            return $this->entityClassNameFinder($associationAdmin, $elements);
+        }
+    }
+
+    /**
+     * get access path to element which works with PropertyAccessor.
+     *
+     * @param string $elementId
+     * @param mixed  $entity
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    public function getElementAccessPath($elementId, $entity)
+    {
+        $propertyAccessor = new PropertyAccessor();
+
+        $initial = preg_replace('#(_(\d+)_)#', '[$2]', implode('_', explode('_', substr($elementId, strpos($elementId, '_') + 1))));
```

Comment:
Intial what? Please describe what is the deal with the underscores in the parameter description (line 253)